x-pack/legacy/plugins/infra/public/containers/metadata/use_metadata.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+import { InfraNodeType } from '../../graphql/types';
+import { InfraMetricLayout } from '../../pages/metrics/layouts/types';
+import { Metadata } from '../../../server/routes/metadata/types';
+import { fetch } from '../../utils/fetch';
```

Comment:
Why do we use our own `fetch` here, instead of `ui/kfetch`?

Reply:
I just re-used what Metrics Explorer used since it worked so well.